code/dropSamples.m

```diff
@@ -1,20 +1,21 @@
 function [XR] = dropSamples(varargin)
-probRemove = (varargin{1});
-if ~iscell(varargin{2})
+probRemove = varargin{1};
 infile = varargin{2};
-if nargin>2
-    outfileR = varargin{3};
-end
-if nargin>3
-    outfileR = varargin{4};
-end
-load(infile);
+% Import X from matfile
+X = infile.X;
+% Get size of X;
+[LX,WX ] = size(infile,'X');
+if ismember('X1',who(infile))
+    % Make sure to remove samples *after* removing zeros (optionally)
+    % Get matrix of logical values corresponding to already removed
+    % samples.
+    XR = infile.X1;
+    for i = 1:LX
+        ind = find(~XR(i,:));
+        ind = ind(rand(1,length(ind))<probRemove);
```

Comment:
Need to use `length(ind)` in order to match the exact same samples as before.  This happens after zeros are removed.